Add REST PyDantic Objects for Task- and ServiceQueue Objects

Lnaden opened the pull request:
## Description
Similar to #130, this PR adds the same GET and POST REST API to the TaskQueue and ServiceQueue.

There are a few oddities here, namely with a few of the methods accepting a variety of inputs, but otherwise this adds client and server side type checking to objects.

The only concern I have is on the `Molecule.dict()` and `.json()` calls in that we cannot reliably recover the `id` field without supplying all other fields, or doing some post-dict addition of the `id` field back in for the `TorsionDrive` Service. I have a quick 2-line workaround for now, so its fine, just something to note (i've added a line-comment below)

I did not touch the QueueManagerHandler object since that is all server side, and may not need validation (or may be more complex) than we need right now. 

## Status
- [x] Ready to go

Lnaden commented on qcfractal/queue/handlers.py:
```diff
@@ -70,12 +93,19 @@ def post(self):
+        for idx, mol in mol_response.data.items():
+            mol_dict = mol.json(as_dict=True)
+            # Ensure ID is present
+            if hasattr(mol, 'id') and mol.id is not None:
```
Workaround the `id` field being dropped on dict conversion

dgasmith commented on qcfractal/interface/client.py:
```diff
@@ -384,13 +388,13 @@ def add_compute(self,
-                    options: str,
-                    molecule_id: str,
+                    options: Union[str, None],
+                    molecule_id: Union[str, Dict[str, Any], List[Union[str, Dict[str, Any]]]],
```
We need to sort this out, the `Dict[str, Any]` should be a Molecule I think?

dgasmith commented:
Looks like there are failures in the examples CLI runs.

Lnaden commented:
I think i fixed the examples now (made the output more helpful)

I cleaned up some of the `import` statements involving `Molecule`, which was imported in 3 places before now

I'm not quite happy with that coverage decrease, I'll look into it

Lnaden commented:
No idea what's going on with CodeCov there, I think it'll be fine though

dgasmith commented:
CodeCov is going down due to incomplete testing I believe. It should patch itself it up if all tests pass.

Lnaden commented:
> CodeCov is going down due to incomplete testing I believe. It should patch itself it up if all tests pass.

So it did. +0.28 overall
